src/components/autosuggest-category/autosuggest-category.tsx

```diff
@@ -0,0 +1,57 @@
+import clsx from "clsx";
+import { UseComboboxPropGetters } from "downshift";
+import * as React from "react";
+import { FC } from "react";
+import { SuggestionsFromQueryStringQuery } from "../../core/dbc-gateway/generated/graphql";
+import { useText } from "../../core/utils/text";
+import { Suggestion } from "../../core/utils/types/autosuggest";
+
+export interface AutosuggestCategoryProps {
+  categoryData: SuggestionsFromQueryStringQuery["suggest"]["result"];
+  getItemProps: UseComboboxPropGetters<Suggestion>["getItemProps"];
+  highlightedIndex: number;
+  textAndMaterialDataLength: number;
+  autosuggestCategoryList: { render: string; type: string }[];
+}
+
+const AutosuggestCategory: FC<AutosuggestCategoryProps> = ({
+  categoryData,
+  getItemProps,
+  highlightedIndex,
+  textAndMaterialDataLength,
+  autosuggestCategoryList
+}) => {
+  const t = useText();
+  return (
+    <li>
+      <ul>
+        {/* eslint-disable react/jsx-props-no-spreading */}
+        {/* The downshift combobox works this way by design (line 45) */}
+        {/* incorrectIndex because in the whole of autosuggest dropdown it is
+          not the correct index for the item. We first need to add the length of
+          items from autosuggest string suggestion to it for it to be accurate. */}
+        {categoryData.map((item, incorrectIndex) => {
+          const index = incorrectIndex + textAndMaterialDataLength;
```

Comment:
Maybe a comment why this calculation is necessary?

Reply:
it is right there :) I'm moving it closer